scripts/check-cryptohasher-symbols.py

```diff
@@ -0,0 +1,65 @@
+'''
+Today if 2 structs/enums use `CryptoHasher` derive and share the same name,
+the current `CryptoHasher` implemtation does not prevent hash input collision.
+This can be a potential vulnerability.
+
+The easiest way is to let aptos developers ensure unique symbol names.
+
+This is a quick and dirty script to help find enum/structs in this repo that
+use `CryptoHasher` derive and share the same name.
+Ultimately it will be done by a rust lint.
+
+How to use:
+    cargo doc --workspace --document-private-items
+    python3 scripts/check-cryptohasher-symbols.py
+'''
+
+from collections import defaultdict
+from pprint import pprint
+import os
+import re
+
+# False positives that needs to be skipped for now.
+whitelisted_symbols = set([
+    'TestAptosCrypto',
+])
+
+ignored_crates = set([
+    'ahash',
+    'aptos_sdk',
+    'seahash',
+    'siphasher',
+])
+
+assert os.path.exists('target/doc')
```

Comment:
Will this directory always be up to date in the CI/CD environment where this script is running? @banool @perryjrandall

Reply:
that should be updated by cargo command i believe target is ignored by .gitignore